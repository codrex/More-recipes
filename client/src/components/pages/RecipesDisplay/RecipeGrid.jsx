import React from 'react';
import PropTypes from 'prop-types';
import RecipeCard from './RecipeCard';
import imageParser from '../../../utils/imageParser';
import Loader from '../../common/Loader';

const RecipeGrid = (props) => {
  const { recipes, loading } = props;
  if (loading) return <Loader loading={loading} />;
  if (recipes.length < 1) return null;

  return (
    <ul className="grid">
      {
        recipes.map(
          recipe => (
            <li className="card-wrapper" key={recipe.id}>
              <RecipeCard
                recipeName={recipe.name}
                category={recipe.category}
                views={recipe.views}
                upVotes={recipe.upVotes}
                downVotes={recipe.downVotes}
                recipeId={recipe.id}
                toggleFav={props.toggleFav}
                isFav={props.isUserFav}
                push={props.history.push}
                image={imageParser(recipe.image)}
              />
            </li>
          )
        )
      }
    </ul>
  );
};

RecipeGrid.defaultProps = {
  loading: false,
  isUserFav: false
};

RecipeGrid.propTypes = {
  recipes: PropTypes.arrayOf(PropTypes.object).isRequired,
  history: PropTypes.objectOf(PropTypes.shape).isRequired,
  loading: PropTypes.bool,
  toggleFav: PropTypes.func.isRequired,
  isUserFav: PropTypes.func,
};

export default RecipeGrid;
