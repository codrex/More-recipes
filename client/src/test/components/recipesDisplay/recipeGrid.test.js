/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import toJson from 'enzyme-to-json';
import RecipeGrid from '../../../components/pages/RecipesDisplay/RecipeGrid';

const recipe = {
  name: 'coco rice',
  category: 'breakfast',
  views: 22,
  upVotes: 22,
  downVotes: 0,
  id: 1
};

describe('Recipes grid component ', () => {
  test(`expected to render RecipeGrid component with 0 RecipeCard component
  when recipes.length is 0`, () => {
      const component = mount(
        <RecipeGrid
          recipes={[]}
          toggleFav={jest.fn()}
          isUserFav={jest.fn()}
          history={{ push: jest.fn() }}
        />
      );
      const tree = toJson(component);
      const recipeCard = component.find('RecipeCard');
      expect(recipeCard.length).toBe(0);
      expect(tree).toMatchSnapshot();
    });

  test(`expected to render RecipeGrid component with 1 RecipeCard component
  when recipes.length is 1`, () => {
      const component = mount(
        <RecipeGrid
          recipes={[recipe]}
          toggleFav={jest.fn()}
          isUserFav={jest.fn()}
          history={{ push: jest.fn() }}
        />
      );
      const tree = toJson(component);
      const recipeCard = component.find('RecipeCard');
      expect(recipeCard.length).toBe(1);
      expect(tree).toMatchSnapshot();
    });
});
