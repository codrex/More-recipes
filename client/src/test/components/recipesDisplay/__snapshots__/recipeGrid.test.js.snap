// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes grid component  render as expected with one recipe 1`] = `
<ul
  className="grid"
>
  <li
    className="card-wrapper"
  >
    <span>
      <div
        className="recipe-card"
      >
        <div
          className="recipe-card-body"
        >
          <img
            className="recipe-card-img"
            height="250"
            src="http://res.cloudinary.com/resycom/image/upload/c_fill,h_250,w_400/recipe-card-placeholder_ta9ikp_iebsf2.jpg"
            width="400"
          />
          <div
            className="recipe-card-underlay"
          >
            <p
              className="lead text-capitalize"
            >
              breakfast
            </p>
            <button
              className="btn recipe-card-underlay-btn"
              data-dismiss="modal"
              data-target=""
              data-toggle=""
              href=""
              id=""
              onClick={[Function]}
              type="button"
            >
              view details
            </button>
          </div>
        </div>
        <div
          className="recipe-card-footer"
        >
          <h4
            className="ellipsis"
          >
            coco rice
          </h4>
          <div
            className="d-flex col-12 no-padding justify-content-between"
          >
            <div
              className="d-flex col-8 no-padding"
            >
              <div
                className="icon-wrapper"
                onClick={[Function]}
                onMouseEnter={[Function]}
                role="button"
                tabIndex="0"
              >
                <i
                  className="icon fa fa-thumbs-up recipe-card-icon "
                  data-for="1upvote"
                  data-target=""
                  data-tip="Total Upvotes: 22"
                  data-toggle=""
                >
                  <span>
                    22
                  </span>
                </i>
                <div
                  className="__react_component_tooltip place-top type-dark "
                  data-id="tooltip"
                >
                  
                </div>
              </div>
              <div
                className="icon-wrapper"
                onClick={[Function]}
                onMouseEnter={[Function]}
                role="button"
                tabIndex="0"
              >
                <i
                  className="icon fa fa-thumbs-down recipe-card-icon "
                  data-for="1downvote"
                  data-target=""
                  data-tip="Total Downvotes: 0"
                  data-toggle=""
                >
                  <span>
                    0
                  </span>
                </i>
                <div
                  className="__react_component_tooltip place-top type-dark "
                  data-id="tooltip"
                >
                  
                </div>
              </div>
              <div
                className="icon-wrapper"
                onClick={[Function]}
                onMouseEnter={[Function]}
                role="button"
                tabIndex="0"
              >
                <i
                  className="icon fa fa-eye recipe-card-icon"
                  data-for="1views"
                  data-target=""
                  data-tip="Total Number of Views: 22"
                  data-toggle=""
                >
                  <span>
                    22
                  </span>
                </i>
                <div
                  className="__react_component_tooltip place-top type-dark "
                  data-id="tooltip"
                >
                  
                </div>
              </div>
            </div>
            <div
              className="icon-wrapper pointer"
              onClick={[Function]}
              onMouseEnter={[Function]}
              role="button"
              tabIndex="0"
            >
              <i
                className="icon fa recipe-card-icon fav-color fav fa-heart-o"
                data-for="1fav"
                data-target=""
                data-tip="Favourite This Recipe"
                data-toggle=""
              >
                <span />
              </i>
              <div
                className="__react_component_tooltip place-top type-dark "
                data-id="tooltip"
              >
                
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </li>
</ul>
`;

exports[`Recipes grid component  render as expected without any recipe 1`] = `null`;
